### PR TITLE
Adding validation e2e tests

### PR DESCRIPTION
**Pull request by lalon4**
- Adding 4 validations e2e tests

### CODE REVIEW

File: discovery-infra/test_infra/utils.py [hunk 1]
```diff
@@ -373,7 +373,11 @@ def wait_till_cluster_is_in_status(
 def is_cluster_in_status(client, cluster_id, statuses):
     log.info("Is cluster %s in status %s", cluster_id, statuses)
     try:
-        return client.cluster_get(cluster_id).status in statuses
+        if client.cluster_get(cluster_id).status in statuses:
+            return True
+        else:
+            log.info(f"Cluster not yet in its required status. "
+                     f"Current status: {client.cluster_get(cluster_id).status}")
```
Comment (tsorya) [inline discovery-infra/test_infra/utils.py:380]: why not to get cluster once and not to work with object? And lets return false from this func and not None

File: discovery-infra/tests/test_validations.py [hunk 2]
```diff
@@ -7,7 +6,16 @@
...
 
+@pytest.mark.validations
 class TestValidations(BaseTest):
+    def _wait_and_kill_installer(self, cluster, nodes, host):
+        # Wait for specific host to be in installing in progress
+        cluster.wait_for_specific_host_status(host=host,
+                                              statuses=[consts.NodesStatus.INSTALLING_IN_PROGRESS])
+        # Kill master installer to simulate host error
+        selected_node = nodes.get_node_from_cluster_host(host)
+        selected_node.kill_podman_container_by_name("assisted-installer")
```
Comment (tsorya) [inline discovery-infra/tests/test_validations.py:17]: maybe came time to have node.kill_installer?

File: discovery-infra/test_infra/utils.py (hunk 1 above)
Comment (lalon4) [reply discovery-infra/test_infra/utils.py]: Fixed

File: discovery-infra/tests/test_validations.py (hunk 2 above)
Comment (lalon4) [reply discovery-infra/tests/test_validations.py]: Fixed

**Comment by lalon4** [comment]
A successful run of validation mark:
https://jenkins-fci-continuous-productization.cloud.paas.psi.redhat.com/view/Assisted-Installer/job/ocp-assisted/1742/testReport/test_validations/TestValidations/

File: discovery-infra/tests/test_validations.py [hunk 3]
```diff
@@ -114,3 +122,89 @@ def test_hosts_and_cluster_max_length(self, cluster, nodes):
...
+        worker_hosts = new_cluster.get_hosts_by_role(role=consts.NodeRoles.WORKER)
+        worker_host1 = worker_hosts[0]
+        worker_host2 = worker_hosts[1]
+        new_cluster.disable_host(host=worker_host1)
+        cluster_details = new_cluster.get_details()
+        self.assert_cluster_validation(cluster_details, "hosts-data", "sufficient-masters-count", "failure")
+        # Disable the second worker node
+        new_cluster.disable_host(host=worker_host2)
+        cluster_details = new_cluster.get_details()
+        new_cluster.wait_for_ready_to_install()
```
Comment (nshidlin) [inline discovery-infra/tests/test_validations.py:153]: Why do you need to wait_for_ready_to_install? If need then you should new_cluster.get_details() after the wait, no?

Comment (lalon4) [reply discovery-infra/tests/test_validations.py]: Your right, its redundant, removed

Comment (lalon4) [reply discovery-infra/tests/test_validations.py]: In second though, no - it should be there. Waiting for cluster validation to pass after disable node

Comment (lalon4) [reply discovery-infra/tests/test_validations.py:153]: when only one node disable - cluster validation fail.
when 2 worker are disabled, than cluster validation should pass, and cluster should turn to ready to install

Comment (nshidlin) [reply discovery-infra/tests/test_validations.py:153]: Okay,
Still shouldn't you new_cluster.get_details() after new_cluster.wait_for_ready_to_install()?

Comment (lalon4) [reply discovery-infra/tests/test_validations.py:152]: Yep, fixed

**Pull request merged**